### DISCUSSION TITLE
Why implement Child and RouterNeighbor structs only in mesh diag and not net diag?

### DISCUSSION THREAD

**Discussion by CodingRays** [openthread/openthread#9302] [Q&A] [answered]
The net diag api allows directly sending diagnostic get requests. But the structs for the new tlv types `Child` and `RouterNeighbor` are only provided by the mesh diag api not the net diag api.

Conceptually shouldn't those be implemented by net diag so that they are available there too and then used by mesh diag?

**Comment by abtink** [comment, marked as answer]
The way we receive and process new Net Diag TLV entries is different from original Net Diag TLVs. This is mainly because new Net Diag TLV entries are list-based, meaning that there can be multiple "children" and/or "router neighbor" entries, each represented by its own TLV. We also have mechanism to indicate we reached end of the list.

Also, to avoid dealing with larger IP messages, the response can be sent over multiple TMF DIAG Answer messages (each containing a subset of Child TLVs). This is handled by the `MeshDiag` class, which makes it easier for users to get the information they need without having to deal with each message individually.
